ha/core/controllers/pcs/node_controller.py

```diff
@@ -336,32 +356,15 @@ def stop(self, node_id: str, timeout: int = -1, **op_kwargs) -> dict:
             ([dict]): Return dictionary. {"status": "", "msg":""}
                 status: Succeeded, Failed, InProgress
         """
-        check_cluster = op_kwargs.get("check_cluster") if op_kwargs.get("check_cluster") is not None else True
         poweroff = op_kwargs.get("poweroff") if op_kwargs.get("poweroff") is not None else False
         storageoff = op_kwargs.get("storageoff") if op_kwargs.get("storageoff") is not None else False
+        # Get the node_name (pvtfqdn) fron nodeid and raise exception if node_id is not valid
+        node_name = self._get_node_name(node_id=node_id)
         try:
-
-            # Get the node_name (pvtfqdn) fron nodeid
-            node_name = self._get_node_name(node_id=node_id)
-            # Raise exception if node_id is not valid
-            self._is_node_in_cluster(node_id=node_name)
-
-            # stop all services running on node with node_id
-            node_status = self._system_health.get_node_status(node_id=node_id).get("status")
-            if node_status == HEALTH_STATUSES.OFFLINE.value:
-                Log.info(f"For stop {node_name}, Node already in offline state.")
-                status = f"Node {node_name} is already in offline state."
-                return {"status": const.STATUSES.SUCCEEDED.value, "output": status, "error": ""}
-            elif node_status == HEALTH_STATUSES.FAILED.value:
-                raise ClusterManagerError(f"Failed to stop {node_name}. node is in {node_status} state.")
-            else:
-                if check_cluster:
-                    # Checks whether cluster is going to be offline if node with node_name is stopped.
-                    res = json.loads(self.check_cluster_feasibility(node_id=node_id))
-                    if res.get("status") == const.STATUSES.FAILED.value:
-                        return res
-                if self.heal_resource(node_name):
-                    time.sleep(const.BASE_WAIT_TIME)
+            stop_status = json.loads(super().stop(node_id, **op_kwargs))
+            if stop_status != None:
+                if stop_status["status"] == const.STATUSES.SUCCEEDED.value or stop_status["status"] == const.STATUSES.FAILED.value:
```

Comment:
Same comment as line #309 above.

Reply:
Added comments in code to avoid confusion.